### PR TITLE
Add a login link to newsletter paywall

### LINKED ISSUE #31857: Add login links to Subscriber paywall
The paywall / subscriber-only wall for newsletters doesn't include a login link, making it much harder to login if you already are a subscriber or paid subscriber.

Let's add the link:

Image

### PR DESCRIPTION
**Pull request by simison**
Resolves https://github.com/Automattic/jetpack/issues/31857

TODO:

- [ ] Better looking email prompt
- [ ] ~Splitting code into separate files in a way that makes sense for subscribe block and paywall separately~. I looked into this but concluded it isn't worth it at least yet; we're adding small amount of JS for the link, but duplicating the rest would actually mean _more_ JS in total for anyone with paywall showing up. That might change if we add some styles/JS into the mix for asking the email, but it's also a good follow-up PR.

## Proposed changes:

* Cleans out translations so that we send full sentences for translations; partial ones are kind of impossible to translate in many languages. I could split that to its own PR though.
* Adds "login" links to the paywall

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Have subscriptions module enabled in Jetpack, or use WP.com simple site
* Connect Stripe and set payments from WP.com `/earn` page or use links at post editor's "Newsletter" sidepanel
* Create two new posts; one for subscribers only, another for paid subscribers only.
* Create a new free subscriber at the site, and paid subscriber
* Try login for both posts with both users

### CODE REVIEW

**Comment by crisbusquets** [comment]
Hi @simison !

Can we use this issue to update the paywall design, so it matches de modal @edanzer was developing?

Screenshot 2023-07-13 at 10 43 20

Figma file (h8tMpJlCqNFemEerU9owHI-fi-3483_150189)

**Comment by simison** [comment]
@crisbusquets  Yeah, sounds good!

I'll try craft something neat looking for the email input but feel free to share quick design here as well.

Ideas:

- Swap paywall content inline to just show input+button block but with button label "Login".
    - Plus some link to "subscribe" instead to return to previous state?
    - image
- A modal same size/look as the payments modal, with the same inputs as above.

Doesn't need to be perfect on first attempt; just good enough. Function over form for this one, because the production experience is poor.

**Comment by crisbusquets** [comment]
> * A modal same size/look as the payments modal, with the same inputs as above.

Just to make sure I understand: this `sign in` link will trigger the flow where the user introduces their email, receive a code, introduce the code and then they're logged in?

If that's the case, I would open that model when they click the `sign-in` link in the paywall.

What do you think?

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php [hunk 1]
```diff
@@ -809,29 +809,32 @@ function maybe_gate_existing_comments( $comment ) {
...
-			<h2 class="wp-block-heading has-text-align-center" style="margin-bottom:var(--wp--preset--spacing--60);font-size:24px">' .
+			<div class="wp-block-group has-border-color has-primary-border-color" style="border-width:1px;border-radius:4px;padding-top:var(--wp--preset--spacing--80);padding-right:var(--wp--preset--spacing--80);padding-bottom:var(--wp--preset--spacing--80);padding-left:var(--wp--preset--spacing--80)">
 
-			/* translators: Newsletter access. Possible values are "paid newsletters" and "subscribers" */
-			sprintf( esc_html__( 'This post is for %s', 'jetpack' ), $access_level )
-
-			. '</h2>
+			<!-- wp:heading {"textAlign":"center","style":{"typography":{"fontSize":"24px"},"spacing":{"margin":{"bottom":"var:preset|spacing|60"}}}} -->
+			<h2 class="wp-block-heading has-text-align-center" style="margin-bottom:var(--wp--preset--spacing--60);font-size:24px">' . esc_html( $access_heading ) . '</h2>
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:829]: I did number of style edits here, but I moved these to a separate PR to keep things easier to review. That should get merged first: https://github.com/Automattic/jetpack/pull/31880

**Comment by lezama** [comment]
Working on a simple version for simple sites here: https://github.com/Automattic/jetpack/pull/32849
